message.go

```diff
@@ -67,29 +68,46 @@ func (mm *MessageManager) Listen(messageType string, handler MessageHandler) Mes
 	return handlerID
 }
 
+// ListenOnce is a convenience wrapper around StopListen() to only listen to a specified message once
+func (mm *MessageManager) ListenOnce(messageType string, handler MessageHandler) {
+	handlerId := MessageHandlerId(0)
+	handlerId = mm.Listen(messageType, func(msg Message) {
+		handler(msg)
+		mm.StopListen(messageType, handlerId)
+	})
+}
+
 // StopListen removes a previously added handler from the listener queue
-func (mm *MessageManager) StopListen(messageType string, handlerId MessageHandlerId) error {
-	// first, we need to find the handler to remove or return an error if that's not possible
+func (mm *MessageManager) StopListen(messageType string, handlerId MessageHandlerId) {
+	if mm.handlersToRemove == nil {
+		mm.handlersToRemove = make(map[string][]MessageHandlerId)
+	}
+	mm.handlersToRemove[messageType] = append(mm.handlersToRemove[messageType], handlerId)
+}
+
+// Will deleted all queued handlers that are scheduled for removal due to StopListen()
+func (mm *MessageManager) clearRemovedHandlers() {
+	for messageType, handlerList := range mm.handlersToRemove {
+		for _, handlerId := range handlerList {
+			mm.removeHandler(messageType, handlerId)
+		}
+	}
```

Comment:
At the end here clear out the handlersToRemove so you're not iterating over things that have already been removed.
`handlersToRemove = make(map[string][]MessageHandlerId)` will do the trick ^_^